GoogleForCreators/web-stories-wp
# CUJ Coverage using Karma reporter

## Summary



## Relevant Technical Choices

Does not include the proposed `Files` column as it seems quite cumbersome to get the original test file name from the test suite. One would have to write a preprocessor to get the file name somehow and deal with the webpack config which puts all tests into one big file.
Also, it seemed nice-to-have anyway, and is not really a requirement.

## To-do

* [x] Implement for dashboard
* [ ] Add table as PR comment?
* [ ] Create a bunch of empty tests

## User-facing changes

N/A

## Testing Instructions

Run `npm run test:karma:edit-story` and verify output.

See console output like this:

```markdown
| **CUJ**                             | **Action**                     | **Completion** |
| ----------------------------------- | ------------------------------ | -------------- |
| Creator can add image/video to page | *\[total\]*                    | 🚨 **0.00%**   |
| Creator can add image/video to page | Search media                   | 🚨 **0.00%**   |
| Creator can add image/video to page | Upload media via upload button | 🚨 **0.00%**   |
| Creator can add image/video to page | Set as background              | 🚨 **0.00%**   |
| Creator can add image/video to page | Filter media                   | 🚨 **0.00%**   |
| Creator can Add and Write Text      | *\[total\]*                    | 🛴 **66.67%**  |
| Creator can Add and Write Text      | Delete textbox                 | 🛴 **66.67%**  |
| *\[total\]*                         | *\[total\]*                    | 🚨 **25.00%**  |
```

...which results in a Markdown table like this:


| **CUJ**                             | **Action**                     | **Completion** |
| ----------------------------------- | ------------------------------ | -------------- |
| Creator can add image/video to page | *\[total\]*                    | 🚨 **0.00%**   |
| Creator can add image/video to page | Search media                   | 🚨 **0.00%**   |
| Creator can add image/video to page | Upload media via upload button | 🚨 **0.00%**   |
| Creator can add image/video to page | Set as background              | 🚨 **0.00%**   |
| Creator can add image/video to page | Filter media                   | 🚨 **0.00%**   |
| Creator can Add and Write Text      | *\[total\]*                    | 🛴 **66.67%**  |
| Creator can Add and Write Text      | Delete textbox                 | 🛴 **66.67%**  |
| *\[total\]*                         | *\[total\]*                    | 🚨 **25.00%**  |

Also, see how the `build/cuj-coverage-edit-story.md` file was created with this output.

---



Fixes #2607
